docs/src/explanations/bundles-and-vecs.md

```diff
@@ -128,48 +128,17 @@ class ModuleProgrammaticMixedVec(x: Int, y: Int) extends Module {
 }
 ```
 
-### A note on `cloneType`
+### A note on `cloneType` (For Chisel < 3.5)
```

Comment:
For something that autoclonetype cannot handle like #1893, should we provide another API to user to ask them implement this on their own? I think a backup API should exist for some corner cases.

Reply:
That's a good question. @jackkoenig , what do you think? Do we anticipate legitimate uses of needing to write your own `def cloneType` in the short and/or long terM?

Reply:
#1893 aside (which we should just fix), you need to write your own `.cloneType` for `Records` as well so the note is still needed but maybe with a warning that this _shouldn't_ be needed except in rare cases.

Reply:
right, but say we say you only need it in rare cases, and then the plugin makes it an error to havea. cloneType method... how do we let the plugin allow/recognize the "rare cases"? Or is it that the plugin should not hit its error scenario if it doesn't recognize the need for the plugin in a certain case (meaning you need your own `def cloneType` for the rare case)?

Reply:
My current thinking is that those "rare cases" are `Records` and bugs. I can't think of any case when one should _want_ to implement their own `cloneType` method, so maybe we just make it an error and fix the bugs that come up. If some use case comes up in the future we can provide an out (perhaps a trait to mixin or a Scala annotation), but I don't think we should do that work unless necessary.